README.md

```diff
@@ -1 +1,415 @@
+	memset(data, 0, sizeof(data));
+	strncpy(data, moisture_str, sizeof(data) - 1);
+	configureTx(dev);
+	ret = lora_send(dev, data, MAX_DATA_LEN);
+	if (ret < 0) {
+		LOG_ERR("LoRa send failed");
+		return 0;
+	}
+
+	LOG_INF("Data sent!");
+
+}
+```
+This function sends data too dev board about soil moisture.
+```c
+void recev_msg(const struct device *dev, uint8_t *data_recive, int16_t rssi, int8_t snr){
+	int len;
+	configureRx(dev);
+	LOG_INF("Recive data");
+	len = lora_recv(dev, data_recive, MAX_DATA_LEN, K_FOREVER, &rssi, &snr);
+	if (len < 0) {
+		LOG_ERR("LoRa receive failed");
+		return 0;
+	}
+
+	LOG_INF("Received data: %s", data_recive);
+}
+```
+This function recevs data from dev board.
+```c
+void gpios_config(const struct device *gpio1, const struct device *gpio2){
+	int ret;
+	ret = gpio_pin_configure(gpio1, 0, GPIO_OUTPUT_ACTIVE);//green
+	if (ret != 0) {
+		return;
+	}
+		ret = gpio_pin_configure(gpio2, 10, GPIO_OUTPUT_ACTIVE);//green
+	if (ret != 0) {
+		return;
+	}
+	if(!device_is_ready(gpio1)){
+		return;
+	}
+	if(!device_is_ready(gpio2)){
+		return;
+	}
+
+
+}
+```
+This function configs gpios outputs for led controling.
+```c
+void adc_config(){
+	int err;
+	for (size_t i = 0U; i < ARRAY_SIZE(adc_channels); i++) {
+		if (!adc_is_ready_dt(&adc_channels[i])) {
+			LOG_INF("ADC controller device %s not ready\n", adc_channels[i].dev->name);
+			return 0;
+		}
+
+		err = adc_channel_setup_dt(&adc_channels[i]);
+		if (err < 0) {
+			LOG_INF("Could not setup channel #%d (%d)\n", i, err);
+			return 0;
+		}
+	}
+}
+```
+This function is used to configs analog to digital converter.
+
+```c
+int adc_val(){
+	uint16_t buf;
+	struct adc_sequence sequence = {
+		.buffer = &buf,
+		.buffer_size = sizeof(buf),
+	};
+	int err;
+	int moisture;
+	for (size_t i = 0U; i < ARRAY_SIZE(adc_channels); i++) {
+		int32_t val_mv;
+		(void)adc_sequence_init_dt(&adc_channels[i], &sequence);
+		err = adc_read_dt(&adc_channels[i], &sequence);
+		if (err < 0) {
+			
+			continue;
+		}
+		if (adc_channels[i].channel_cfg.differential) {
+			val_mv = (int32_t)((int16_t)buf);
+		} else {
+			val_mv = (int32_t)buf;
+		}
+		err = adc_raw_to_millivolts_dt(&adc_channels[i],&val_mv);
+		int val_int = (int)val_mv;
+		moisture = mapSoilMoisture(val_int, DryValue, WetValue, 0, 100);
+
+	}
+	return moisture;
+}
+```
+This function returns values of moisture in range 0-100.
+```c
+void set_led(uint8_t *data_recive, const struct device *gpio1, const struct device *gpio2){
+	int ret;
+	int SetLed = atoi(data_recive);
+	if (SetLed == 1) {
+		
+		ret = gpio_pin_set_raw(gpio2, 10, 0);
+		ret = gpio_pin_set_raw(gpio1, 0, 1);	
+			
+	} else {
+		
+		ret = gpio_pin_configure(gpio1, 5,GPIO_OUTPUT_ACTIVE); //red
+		if (ret != 0) {
+			return;
+		}
+		
+		ret = gpio_pin_set_raw(gpio1, 0, 0);
+		ret = gpio_pin_set_raw(gpio2, 10, 1);
+			
+	} 
+}
+```
+This function sets color of leds depending on the values send from dev board.
+## Finishing
```

Comment:
I don't like "Finishing" as the name of the section but I don't have any idea how to name it. @styczen @adamkrawczyk

Reply:
Maybe `Conclusion` ?